src/boot/bootctl-status.c

```diff
@@ -415,96 +416,219 @@ int verb_status(int argc, char *argv[], void *userdata) {
-                printf(" Firmware Arch: %s\n", get_efi_arch());
-                printf("   Secure Boot: %s%s%s",
-                       IN_SET(secure, SECURE_BOOT_USER, SECURE_BOOT_DEPLOYED) ? ansi_highlight_green() : ansi_normal(),
-                       enabled_disabled(IN_SET(secure, SECURE_BOOT_USER, SECURE_BOOT_DEPLOYED)),
-                       ansi_normal());
-
-                if (secure != SECURE_BOOT_DISABLED)
-                        printf(" (%s)\n", secure_boot_mode_to_string(secure));
-                else
-                        printf("\n");
-
                 s = tpm2_support();
-                printf("  TPM2 Support: %s%s%s\n",
-                       FLAGS_SET(s, TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER) ? ansi_highlight_green() :
-                       (s & (TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER)) != 0 ? ansi_highlight_red() : ansi_highlight_yellow(),
-                       FLAGS_SET(s, TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER) ? "yes" :
-                       (s & TPM2_SUPPORT_FIRMWARE) ? "firmware only, driver unavailable" :
-                       (s & TPM2_SUPPORT_DRIVER) ? "driver only, firmware unavailable" : "no",
-                       ansi_normal());
-
-                k = efi_measured_uki(LOG_DEBUG);
-                if (k > 0)
-                        printf("  Measured UKI: %syes%s\n", ansi_highlight_green(), ansi_normal());
-                else if (k == 0)
-                        printf("  Measured UKI: no\n");
-                else {
-                        errno = -k;
-                        printf("  Measured UKI: %sfailed%s (%m)\n", ansi_highlight_red(), ansi_normal());
-                }
-
-                k = efi_get_reboot_to_firmware();
-                if (k > 0)
-                        printf("  Boot into FW: %sactive%s\n", ansi_highlight_yellow(), ansi_normal());
-                else if (k == 0)
-                        printf("  Boot into FW: supported\n");
-                else if (k == -EOPNOTSUPP)
-                        printf("  Boot into FW: not supported\n");
-                else {
-                        errno = -k;
-                        printf("  Boot into FW: %sfailed%s (%m)\n", ansi_highlight_red(), ansi_normal());
-                }
-                printf("\n");
-
-                printf("%sCurrent Boot Loader:%s\n", ansi_underline(), ansi_normal());
-                printf("      Product: %s%s%s\n", ansi_highlight(), strna(loader), ansi_normal());
-
-                for (size_t i = 0; i < ELEMENTSOF(loader_flags); i++)
-                        print_yes_no_line(i == 0, FLAGS_SET(loader_features, loader_flags[i].flag), loader_flags[i].name);
+                const char *tpm2_support_str = FLAGS_SET(s, TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER) ? "yes" :
+                                                       (s & TPM2_SUPPORT_FIRMWARE) ? "firmware only, driver unavailable" :
```

Comment:
The purpose of --json is to show machine-readable output, but these strings are just a copy of the normal output. The values need to be changed to something else that is appropriate for the use case, here and everywhere else

Reply:
I quite do not understand this statement, kindly help clarify.

Reply:
"firmware only, driver unavailable" is not something that is consumed by a program, is something for a person to read, so in case of json output there needs to be an appropriate output not just in formatting but also with regards to its content

Reply:
I do not output "firmware only, driver unavailable" in the json. If I do not get this right, can you help with a sample. Thank you.

Reply:
yes, it does, because the tpm2_support_str variable is used in a json payload below - look at how other existing commands generate json output, it needs to be machine readable entries, not the same log messages as json payloads

Reply:
I check through other implementation, the key is right and I understand json can have spaces in it values. 

I think I quite do not understand "not the same log messages as json payloads" so, it is making it hard for me to figure it out even after checking other implementions.

Reply:
The purpose of json output is to provide machine-readable data, not free-form text. "firmware only, driver unavailable" is not data, it's free-form text that looks like a log message. The equivalent in data format would be something like `"firmware": true, "driver": false` or something along those lines